sycl/test/esimd/dpas.cpp

```diff
@@ -58,7 +58,7 @@ SYCL_ESIMD_FUNCTION SYCL_EXTERNAL void old_func() {
   constexpr int N_pvc = 16;
   constexpr int N_dg2 = 8;
 
-  // CHECK: define dso_local spir_func void @_Z8old_funcv()
+  // CHECK-LABEL: define dso_local spir_func void @_Z8old_funcv()
```

Comment:
I think this is not the case where CHECK-LABEL should be used. The line here is a call, not entry or loop label that ends with ":". The problems that you were trying to fix were caused by copy-paste error that is finally fixed.
Do you agree?

Reply:
> I think this is not the case where CHECK-LABEL should be used. The line here is a call, not entry or loop label that ends with ":". The problems that you were trying to fix were caused by copy-paste error that is finally fixed. Do you agree?

No. The problem was hidden because of missing `CHECK-LABEL`. It only canbe reproduced with `CHECK-LABEL`. Without them there is no guarantee that wrong matching will not happen again. In this context, begin/end act as "barriers" for matching. There is no guideline that such can be used only against BB labels.

Reply:
Ok, that is BTW, why I asked why not to remove old::dpas(), so the checks for xmx::dpas would not get mixed with old::dpas.

Reply:
> Ok, that is BTW, why I asked why not to remove old::dpas(), so the checks for xmx::dpas would not get mixed with old::dpas.

That is not what was happening. Each test function was inlined into the kernel and remained in the module. There were two copies of each function body. Matching happened erroneously with the following copy. The problem will remain even if I remove old::dpas.